tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -192,9 +202,10 @@ def parse_datafile_lines(datafile, program_type, section, is_encrypted):
         if trailer_errors is not None:
             logger.debug(f"{len(trailer_errors)} trailer error(s) detected for file " +
                          f"'{datafile.original_filename}' on line {line_number}.")
-            errors['trailer'] = trailer_errors
-            unsaved_parser_errors.update({"trailer": trailer_errors})
-            num_errors += len(trailer_errors)
```

Comment:
Are we going to reuse `generate_trailer_errors()`? I didn't see it used in the unit tests. It just seems odd to replace 3 lines with 13 lines to perform the same actions. Did we make the functional to be more inline with other functions?

Reply:
We need the function to optionally enable/disable the trailer errors because it has an `if` statement in it. Leaving it in `parse_datafile_lines` made the function too complex.

Reply:
the ```parse_datafile_lines``` function needs some love and cleanup